frame/staking/src/lib.rs

```diff
@@ -1626,16 +2021,203 @@ impl<T: Trait> Module<T> {
 				Forcing::ForceNew => ForceEra::kill(),
 				Forcing::ForceAlways => (),
 				Forcing::NotForcing if era_length >= T::SessionsPerEra::get() => (),
-				_ => return None,
+				_ => {
+					// not forcing, not a new era either. If final, set the flag.
+					if era_length + 1 >= <T::SessionsPerEra as Get<SessionIndex>>::get() {
```

Comment:
if the above get is correctly infered I would be very surprised tis on  is not, though I only tested on my nightly 
```suggestion
					if era_length + 1 >= T::SessionsPerEra::get() {
```